python/ray/_private/services.py

```diff
@@ -785,10 +785,17 @@ def preexec_fn():
             creationflags=CREATE_SUSPENDED if win32_fate_sharing else 0,
         )
     else:
+        # `os.dup2` only takes fd instead of file handler
+        if stdout_file is not None:
+            if (hasattr(stdout_file, "fileno")):
```

Comment:
According to PEP https://peps.python.org/pep-0020/ , the best practice is to use flatting instead of nested `if`.

Reply:
good suggestion!